src/components/Calendar/Section.js

```diff
@@ -36,28 +43,62 @@ export const styles = {
     justifyContent: 'space-between',
     flexWrap: 'wrap',
   },
+  modal: {
+    position: 'absolute',
+    width: '450px',
+    outline: 'none',
+    top: '12.5%',
+    left: '12.5%',
+    padding: '10px',
+  },
 };
 
-function Section({ classes, section }) {
-  return (
-    <Paper className={classes.paper}>
-      <div className={classes.container}>
-        <div className={classes.header}>
-          <Typography variant="caption" className={classes.text}>
-            {getFormattedClassEvent(section.event)}
-          </Typography>
+class Section extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showDialog: false,
+    };
+    this.toggleDialog = this.toggleDialog.bind(this);
+  }
+
+  toggleDialog() {
+    const { showDialog } = this.state;
+    this.setState({
+      showDialog: !showDialog,
```

Comment:
Whenever you're updating state that depends on the current state, use the functional form for `setState` since [`setState` is asynchronous](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous). So that would look like
```javascript
this.setState(state => ({ showDialog: !state.showDialog }));
```